core/dnsserver/doh/https.go

```diff
@@ -0,0 +1,25 @@
+package doh
+
+import (
+	"io/ioutil"
+	"net/http"
+
+	"github.com/miekg/dns"
+)
+
+// MimeType is the DoH mimetype that should be used.
+const MimeType = "application/dns-udpwireformat"
+
+// RequestToMsg extra the dns message from the request body.
+func RequestToMsg(req *http.Request) (*dns.Msg, error) {
```

Comment:
note that this only handle POST.
For GET use case, the content-type should be taken from the query parameter `ct` and the dns payload will be base64safeurl encoded (with trailing `=` removed) in the `dns` query parameter. Maybe worth a FIXME/TODO comment.